src/shared/pager.c

```diff
@@ -165,25 +165,34 @@ int pager_open(PagerFlags flags) {
                 }
 
                 /* People might invoke us from sudo, don't needlessly allow less to be a way to shell out
-                 * privileged stuff. */
-                r = getenv_bool("SYSTEMD_LESSSECURE");
-                if (r == 0) { /* Remove env var if off */
-                        if (unsetenv("LESSSECURE") < 0) {
-                                log_error_errno(errno, "Failed to uset environment variable LESSSECURE: %m");
-                                _exit(EXIT_FAILURE);
-                        }
-                } else {
-                        /* Set env var otherwise */
-                        if (r < 0)
-                                log_warning_errno(r, "Unable to parse $SYSTEMD_LESSSECURE, ignoring: %m");
+                 * privileged stuff. If the user set $SYSTEMD_PAGERSECURE, trust their configuration of the
+                 * pager. If they didn't, use secure mode when under sudo. If $SYSTEMD_PAGERSECURE wasn't
+                 * explicitly set, only use the pager we know to be good. */
+                int secure = getenv_bool("SYSTEMD_PAGERSECURE");
+                bool trust_pager = secure >= 0;
```

Comment:
initializing this here is kinda pointless, given that in the < 0 case you override this anyway in all all cases. maybe move the "false" initialization into a final "else" branch of the "if" check below?

Reply:
Do you mean `secure`? I initialize it first becaseu I want to set `trust_pager`... Maybe there's a nicer way to do this without repeated calls, but I don't see it.